### PR TITLE
Lab5 - Stringer

### LINKED ISSUE #411: Lab 5 - Stringer
- Create an executable go program in directory `05_stringer/USERNAME`
- Make the `IPAddr` type implement `fmt.Stringer` to print the address as a dotted quad
- Find hints at [tour of go exercise: stringers](https://tour.golang.org/methods/18)
- Call `fmt.Println(IPAddr{127, 0, 0, 1})` in `main` to print:

```
127.0.0.1
```

### PR DESCRIPTION
**Pull request by mohitnag**
Fixes #411 

Review of colleague's PR #404

#### Changes proposed in this PR:

- Implements fmt.Stringer interface for IPAddr type
- Adds tests

### CODE REVIEW

File: 05_stringer/mohitnag/main_test.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+func TestIpaddr(t *testing.T) {
+	testData := []struct {
+		Scenario string
+		input    IPAddr
+		expected string
+	}{
+		{"Scenario One", [4]byte{127, 0, 0, 1}, "127.0.0.1"},
+		{"Scenario two", [4]byte{}, "0.0.0.0"},
+		{"Scenario three", [4]byte{0, 0, 0, 10}, "0.0.0.10"},
```
Comment (alfredxiao) [inline 05_stringer/mohitnag/main_test.go]: Would be better to test edge cases like 255.255.255.255

Comment (alfredxiao) [reply 05_stringer/mohitnag/main_test.go]: Test the case like [4]byte{10}

File: 05_stringer/mohitnag/main_test.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+}
+func TestMain(t *testing.T) {
+	assert := assert.New(t)
+	var buf bytes.Buffer
+	out = &buf
+	main()
+	expected := "127.0.0.1"
+	actual := buf.String()
+	actual = strings.Replace(actual, "\n", ",", -1)
+	actual = strings.TrimRight(actual, ",")
```
Comment (alfredxiao) [inline 05_stringer/mohitnag/main_test.go]: No need to `Replacer` or `TrimRight`?

File: 05_stringer/mohitnag/main_test.go (hunk 1 above)
Comment (mohitnag) [reply 05_stringer/mohitnag/main_test.go]: Done

File: 05_stringer/mohitnag/main_test.go (hunk 2 above)
Comment (mohitnag) [reply 05_stringer/mohitnag/main_test.go]: Done

File: 05_stringer/mohitnag/main_test.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIpaddr(t *testing.T) {
+	testData := []struct {
+		Scenario string
+		input    IPAddr
+		expected string
+	}{
+		{"Scenario One", [4]byte{127, 0, 0, 1}, "127.0.0.1"},
```
Comment (rokane) [inline 05_stringer/mohitnag/main_test.go]: I guess this comes down to personal preference and how you want your tests to output when running / failing, but I think it is nicer to give a descriptive name rather than "Scenario x".

Something more like "localhost", "empty", "Two Values" etc to give it a little bit more when outputting a failure. Can you please update this?

**Review by rokane — CHANGES_REQUESTED** [review]
Looking really good from a code perspective, I left one comment to address. Are you also able to squash the commits in this pull request down so that you only have 1-2? This was a quirement for labs 1-5.

File: 05_stringer/mohitnag/main_test.go (hunk 3 above)
Comment (mohitnag) [reply 05_stringer/mohitnag/main_test.go]: Done

**Comment by mohitnag** [comment]
Squashed commits down to 2

**Review by rokane — APPROVED** [review]
Well done @mohitnag and thanks for squashing those commits! Keep it up :)
